### DISCUSSION TITLE
Polyline class does not have a UniquePolylineProps interface

### DISCUSSION THREAD

**Discussion by gloriousjob** [fabricjs/fabric.js#10075] [Q&A] [answered]
Most of the classes in Fabric (e.g., Rect) appear to use a Unique*Props interface to list out the properties that the class provides (e.g., rx, ry in Rect). But Polyline seems to separate the points out of Props. Is this intentional since points is a constructor parameter? I'm wondering if my subclass should be trying to include points in the Props or if it should just extend FabricObjectProps only.

**Comment by asturur** [comment, marked as answer]
What is the issue you are facing?
As long as you are able to clearly see the points in the type when the object is serialized and if you can pass them as first argument of the constructor, that should be good.

If you prefer to move the position of points from first argument to a generic key in the object you should also write your own fromObject method.

**Reply by gloriousjob** [reply]
Thanks! Just wanted to verify the proper way to subclass this. I won't worry about pushing points into Props then. Also wanted to be sure this was intentional in the Polyline class, which it is.

**Reply by asturur** [reply]
The polyline clas was built in 2010 :) I never removed points from where they are becuase i think if is not giving an extra benefit, why bother breaking existing code.
It is true that there are exceptions:
Text, Polygon, Polyline, Path, Image do have a special parameter, while Circle, Triangle, Ellipse, Rect do not have.
